### DISCUSSION TITLE
How to change the sequence of statement in a continuous binary expression

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1761] [Q&A]
Hi Langium team, I have a continuous binary expression, now the grouping of statement is as shown in the black background. 
But I want to change it to the one in the white background. How can I achieve this goal?
 
![image](https://github.com/user-attachments/assets/dd6c0de7-a478-4f9d-ade8-731613d8c1a4)

![image](https://github.com/user-attachments/assets/c07f1d6f-f195-4652-ab56-d24401e08341)

**Comment by msujew** [comment]
Hey @durianwaffle,

you can give an element in your expression chain a higher precedence by putting it closer to the end of the chain. In this case, you need to put `and` closer to the end of the chain than `=`.

**Reply by Lotes** [reply]
[This link](https://www.typefox.io/blog/parsing-expressions-with-xtext/) might help as well with more details. It is for Xtext, but the principle is the same :-) . Operator precedence is a classical problem in compiler theory and is normally solved by a special-shaped grammar.

**Reply by aabounegm** [reply]
@durianwaffle I had a similar problem and followed the instructions from the link sent by @Lotes. For reference, here is what I did: https://github.com/aabounegm/stella-experiment/commit/1e1e222f8312d512dad1a73420d7b87ad80eb329
Thought it might help

**Reply by durianwaffle** [reply]
Hi @msujew, may I know what's the version of langium the playground is using?
I found that the sequence for our grammar is correct in the playground.
But not working in langium version 3.0.

**Reply by spoenemann** [reply]
It should be 3.0.0, too:

https://github.com/eclipse-langium/langium-website/blob/39999f513d9fa449a7728b72ee38da37416437a1/hugo/package.json#L42

**Reply by durianwaffle** [reply]
Okay, thank you @spoenemann !
